File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/generator/Generator.java
package tasks.generator;

import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Text;

@JavaTaskInfo(generator = true)
public class Generator extends JavaTask {

	@OutputPortInfo(name = "Out", dataType = Text.class)
	public OutputPort out;

	@Override
	public void generate() {
		out.putData(new Text("Hello World"));
		sleep(3000);
	}
}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/generator/Generator.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Text import Text

@PythonTaskInfo(generator=True)
class Generator(PythonTask):

    def __init_ports__(self):
        self.out = OutputPort(name="out", data_type=Text)

    def generate(self):
        self.out.put_data(Text("Hello World"));
        self.sleep(3000);

